src/test/ui/parser/recover-from-bad-variant.stderr

```diff
@@ -0,0 +1,23 @@
+error: expected type, found `3`
+  --> $DIR/recover-from-bad-variant.rs:7:26
+   |
+LL |     let x = Enum::Foo(a: 3, b: 4);
+   |                          ^ expecting a type here because of type ascription
+   |
+   = note: type ascription is a nightly-only feature that lets you annotate an expression with a type: `<expr>: <type>`
```

Comment:
As a follow up to this PR, could you point out the feature gate here?

Reply:
@Centril, do we want to nudge people towards enabling type ascription, particularly when the most common case for this error is caused by incorrect syntax, not an attempt to use the nightly feature?

Reply:
@estebank I don't think it wold be particularly invasive to point it out... It could be as simple as:
```
    = note: #![feature(type_ascription)] is a nightly-only feature that lets you annotate an expression with a type: `<expr>: <type>`

Reply:
https://github.com/rust-lang/rust/issues/61325